src/main/java/Duke.java

```diff
@@ -1,10 +1,234 @@
+                throw new DukeException();
+            }
+            int listIndex = Integer.parseInt(line.substring(DELETE_TASK_INDEX)) - 1;
+            if (listIndex >= tasks.size()) {
+                throw new DukeException();
+            } else if (tasks.get(listIndex) == null){
+                throw new DukeException();
+            }
+            else {
+                System.out.println(lineCutOff);
+                System.out.println("Noted. I've removed this task: ");
+                System.out.println(tasks.get(listIndex).toString());
+                System.out.println(lineCutOff);
+                tasks.remove(tasks.get(listIndex));
+                System.out.println("Now you have " + tasks.size() + " items in the list.");
+                System.out.println(lineCutOff);
+            }
+        } catch (DukeException e) {
+            System.out.println(lineCutOff);
+            System.out.println("☹ OOPS!!! There is no such task.");
+            System.out.println(lineCutOff);
+        }
+    }
+
+    public static void printList(ArrayList<Task> tasks) {
+        System.out.println(lineCutOff);
+        if (tasks.size() == 0) {
+            tasks.add(null);
+        }
+        try {
+            if (tasks.get(0) == null) {
+                throw new DukeException();
+            } else {
+                System.out.println("Here are the tasks in your list:");
+                int i = 1;
+                for (Task task: tasks) {
+                    System.out.println(i + ". " + task.toString());
+                    if (i == tasks.size()) {
+                        break;
+                    }
+                    i++;
+                }
+            }
+        } catch (DukeException e) {
+            System.out.println("☹ OOPS!!! Your list seems empty");
+        }
+        System.out.println(lineCutOff);
+    }
+
+    public static void setTasks(String line) {
+        if(tasks.size() == 0)
+            tasks.add(null);
+        if (tasks.get(tasks.size() - 1) != null) {
+            tasks.add(taskType(line));
+        } else if (tasks.get(tasks.size() - 1) == null) {
+            tasks.set(tasks.size() - 1, taskType(line));
+        }
+        if (tasks.get(tasks.size() - 1) != null) {
+            printTask(tasks.get(tasks.size() - 1));
+        }
+    }
+
+
     public static void main(String[] args) {
         String logo = " ____        _        \n"
                 + "|  _ \\ _   _| | _____ \n"
                 + "| | | | | | | |/ / _ \\\n"
                 + "| |_| | |_| |   <  __/\n"
                 + "|____/ \\__,_|_|\\_\\___|\n";
         System.out.println("Hello from\n" + logo);
+        System.out.println(lineCutOff + "\nHello! I'm Duke\nWhat can I do for you?" + "\n" + lineCutOff);
+
+
+        //tasks.add(null);
+        readFromFile(FILE_PATH);
+        Scanner in = new Scanner(System.in);
+        String line = in.nextLine();
+
+        while (!(line.matches("Bye") ||line.matches("bye"))) {
```

Comment:
It looks like there is a small bug where using "bYE" will not cause the program to exit. Perhaps consider using `line.lower().matches("bye")` so that it can catch all cases of bye instead.